tests/unit/internal-session-test.js

```diff
@@ -726,4 +727,19 @@ describe('InternalSession', () => {
       });
     });
   });
+
+  describe('multiple internal sessions', () => {
+    let session2;
+
+    beforeEach(() => {
+      session2 = InternalSession.create({ store, container });
+    });
+
+    it('uses separate content objects', () => {
+      Ember.run(function() {
+        session.set('content.secret', 'I see dead people.');
+      });
+      expect(session2.get('content.secret')).to.be.undefined;
+    });
+  });
```

Comment:
Okay. Fixed this, but it actually needs to be `equal` not `eql` because the two separate empty objects are trivially deep-equal, so I use `equal`. I thought explicitly setting something and testing for the result was more illustrative of the desired behavior rather than proof of implementation. But your solution is fine too.